jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/JAXRSCommonClient.java

```diff
@@ -0,0 +1,1098 @@
+    //TestUtil.logTrace("[JAXRSCommonClient] Contextroot set at " + root);
+    _contextRoot = root;
+  }
+
+  public String getContextRoot() {
+    //TestUtil.logTrace("[JAXRSCommonClient]getContextRoot");
+    return _contextRoot;
+  }
+
+  /**
+   * <code>setup</code> is by the test harness to initialize the tests.
+   * 
+   * @param args
+   *          a <code>String[]</code> value
+   * @param p
+   *          a <code>Properties</code> value
+   * @exception Fault
+   *              if an error occurs
+   */
+  public void setup(String[] args, Properties p)   {
+    //TestUtil.logTrace("setup method JAXRSCommonClient");
+
+    String hostname = p.getProperty(SERVLETHOSTPROP);
+    String portnum = p.getProperty(SERVLETPORTPROP);
+    String tshome = p.getProperty(TSHOME);
+
+    assertTrue(!isNullOrEmpty(hostname),
+        "[JAXRSCommonClient] 'webServerHost' was not set in the build.properties.");
+    _hostname = hostname.trim();
+    assertTrue(!isNullOrEmpty(portnum),
+        "[JAXRSCommonClient] 'webServerPort' was not set in the build.properties.");
+    _port = Integer.parseInt(portnum.trim());
+    assertTrue(!isNullOrEmpty(tshome),
+        "[JAXRSCommonClient] 'tshome' was not set in the build.properties.");
+    _tsHome = tshome.trim();
+
+    //TestUtil.logMsg("[JAXRSCommonClient] Test setup OK");
+  }
+
+  /**
+   * <code>cleanup</code> is called by the test harness to cleanup after text
+   * execution
+   * 
+   * @exception Fault
+   *              if an error occurs
+   */
+  public void cleanup()   {
+    //TestUtil.logMsg("[JAXRSCommonClient] Test cleanup OK");
+  }
+
+  /*
+   * protected methods
+   * ========================================================================
+   */
+  /**
+   * <PRE>
+   * Invokes a test based on the properties
+   * stored in TEST_PROPS.  Once the test has completed,
+   * the properties in TEST_PROPS will be cleared.
+   * </PRE>
+   * 
+   * @ 
+   *           If an error occurs during the test run
+   */
+  protected void invoke()   {
+    //TestUtil.logTrace("[JAXRSCommonClient] invoke");
+    try {
+      _testCase = new WebTestCase();
+      setTestProperties(_testCase);
+      //TestUtil.logTrace("[JAXRSCommonClient] EXECUTING");
+      if (_useSavedState && _state != null) {
+        _testCase.getRequest().setState(_state);
+      }
+      if (_redirect != false) {
+        //TestUtil.logTrace("##########Call setFollowRedirects");
+        _testCase.getRequest().setFollowRedirects(_redirect);
+      }
+      _testCase.execute();
+      if (_saveState) {
+        _state = _testCase.getResponse().getState();
+      }
+    } catch (TestFailureException tfe) {
+      Throwable t = tfe.getRootCause();
+      if (t != null) {
+        //TestUtil.logErr("Root cause of Failure: " + t.getMessage(), t);
+      }
+      //throw new Fault("[JAXRSCommonClient] " + _testName
+      //    + " failed!  Check output for cause of failure.", tfe);
+    } finally {
```

Comment:
I think you will need to throw an exception here. Otherwise test failures will be reported as passing.
```suggestion
    } catch (TestFailureException tfe) {
      Throwable t = tfe.getRootCause();
      if (t != null) {
        //TestUtil.logErr("Root cause of Failure: " + t.getMessage(), t);
        if (t instanceof RuntimeException) {
          throw (RuntimeException) t;
        } else if (t instanceof Error) {
          throw (Error) t;
        } else {
          throw new RuntimeException(t);
        }
      }
      if (Boolean.getBoolean("tck.debug")) {
        tfe.printStackTrace();
      }
      throw new RuntimeException("Unexplained test failure", tfe);
      //throw new Fault("[JAXRSCommonClient] " + _testName
      //    + " failed!  Check output for cause of failure.", tfe);
    } finally {
```

This still isn't great, because `WebTestCase` often just throws the `TestFailureException` with no caused-by exception.  Ideally there would be some indication of what failed - like an `AssertionFailedException` indicating what was the expected vs actual results were.

Reply:
Added the Fault (extends Exception) class back after removing it earlier.